backend/open_webui/models/models.py

```
@@ -128,147 +110,421 @@ class ModelUserResponse(ModelModel):
+            db=db,
+            query=query,
+            DocumentModel=Model,
+            filter=filter,
+            resource_type="model",
+            permission=permission,
+        )
+
+    def search_models(
+        self,
+        user_id: str,
+        filter: dict = {},
```

Ruff (0.15.1)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)